src/yang_types.erl

```diff
@@ -703,11 +703,23 @@ enumeration_type_spec_fun({parse, _Val, _Pos},
     {undefined, Ctx};
 enumeration_type_spec_fun({parse, Val, _Pos},
                           #type{type_spec =
-                                  #enumeration_type_spec{enums = Enums}} = Type,
+                                  #enumeration_type_spec{all_enums = AllEnums, all_enum_stmts = AllStmts}} = Type,
                           _M, _Ctx) ->
-    case lists:keyfind(?b2a(Val), 1, Enums) of
+    case lists:keyfind(?b2a(Val), 1, AllEnums) of
         {_, IntVal} ->
-            {ok, IntVal};
+            case lists:keyfind(?b2a(Val), 2, AllStmts) of
```

Comment:
I think that this is not the right place to do this test - this function parses a string into an enum.  In this module it is only used to parse default values, which is why this works.  But the function is more generic than that - it isn't called "{parse_default_value", just "{parse".  It is also exported which means that it is part of the API.

Instead, I think the right place is in "derive".  The check must be different; you need to check if a default statement is present in the TypeS, and if so, find the corresponing enum statement and report an error if it has a if-feature statement.

Ditto for bits.

Reply:
I do this test according to the current logic - "Val" which is a default value is used to check whether the default value is defined, I just add a check about the default value without a `if-feature` statement, so I think this may be the right position.

Follow as your tips,  I reread the code about parsing the types of enumeration/bits, then debug it. If this test is placed in "derive" here, https://github.com/mbj4668/yanger/blob/8adf023ecab17abb085879b1ed233cce3f1f7302/src/yang_types.erl#L675-L682  there is no default statement in "TypeS",   it seems no way to get 
the default value, of course, I am not sure that I exactly understand  what you mean.